### PR TITLE
Fix building libraries with Ninja.

### LINKED ISSUE #45763: Cannot build runtime repo on Windows with ninja
Clean clone of the runtime repo, Windows 10, latest VS dogfood, running `build.cmd -ninja` from root:

```console
cl : Command line error D8016 : '/std:c++latest' and '/std:c11' command-line options are incompatible
cl : Command line warning D9025 : overriding '/TP' with '/TC'
```

while building `src/coreclr/pal/src/libunwind`

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Fixes #45763

You need to delete the intermediate files and let CMake regenerate them from scratch for this fix to work.

### CODE REVIEW

File: src/libraries/Native/Windows/CMakeLists.txt [hunk 1]
```diff
@@ -1,7 +1,8 @@
 cmake_minimum_required(VERSION 3.14.5)
+cmake_policy(SET CMP0091 NEW)
```
Comment (akoeplinger) [inline src/libraries/Native/Windows/CMakeLists.txt:2]: we're wrapping this in a cmake version check elsewhere:
https://github.com/dotnet/runtime/blob/05cd29c81434aae62d0cb680768d0229c8da0d2d/src/installer/corehost/CMakeLists.txt#L3-L5

Comment (jkoritzinsky) [reply src/libraries/Native/Windows/CMakeLists.txt:2]: Our minimum version of CMake for Windows is 3.16.0 and this CMake entry-point is only used on Windows, so we shouldn't need the version check.

**Comment by vitek-karas** [comment]
I verified that with this change I can build with ninja fully without errors (starting with clean repo).

File: src/libraries/Native/Windows/CMakeLists.txt (hunk 1 above)
Comment (akoeplinger) [reply src/libraries/Native/Windows/CMakeLists.txt:2]: Ok but then we should bump the `cmake_minimum_required` line above to 3.16.0.

File: src/libraries/Native/Windows/CMakeLists.txt [hunk 2]
```diff
@@ -1,7 +1,8 @@
 cmake_minimum_required(VERSION 3.14.5)
```
Comment (jkoritzinsky) [inline src/libraries/Native/Windows/CMakeLists.txt:1]:
```suggestion
cmake_minimum_required(VERSION 3.16)
```

**Review by akoeplinger — APPROVED** [review]

**Comment by jkoritzinsky** [comment]
Test failures are https://github.com/dotnet/runtime/issues/47374 and https://github.com/dotnet/runtime/issues/47474

**Pull request merged**